mantidproject/mantid
# Make curves dropdown list in figure options match legend order

Report to: Steve King

**Description of work.**
This PR makes it so that the dropdown list for selecting a curve in the Curves tab of the figure options has the curves listed in the same order as they appear in the plot legend so that it's easier to choose the correct curve you wish to edit.

**To test:**
Run
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt

a = CreateSampleWorkspace()
b = CreateSampleWorkspace()
c = CreateSampleWorkspace()

fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
ax.plot(c, specNum=1)
ax.plot(b, specNum=1)
ax.plot(a, specNum=1)
ax.legend()
plt.show()
```
Open figure options, go to the curves tab, and check that the order of the curves in the "Select a curve" dropdown list is the same as in the legend.
Mess around with changing a curve's properties (this will turn the Line2D into an ErrorbarContainer), close and reopen the figure options and check that the curve order is always correct.

Fixes #28901 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
